data/blog: add on the node blog post (#299)
tailscale-dev/tailscale-dev: data/blog/on-the-node-while-on-the-road.mdx

---
title: On the Node While on the Road with Tailscale
date: 2023-08-17
authors: ['claire']
tags:
  - exit node
summary: Circumventing sketchy hotel wifi using an exit node.
images: ['/images/on-the-node-while-on-the-road/social.png']
---

Back in April for RSA 2023, a bunch of Tailscalars were staying at a hotel near the conference center. I wanted to get some work done, and tried pulling code from Github using ssh. An ominous message appeared on my screen.

<Figure
  src="/images/on-the-node-while-on-the-road/someone-is-doing-something-nasty.png"
  width={872}
  height={443}
  alt={`Someone is doing something nasty!`}
/>

To be honest, I have never encountered this before. A fingerprint is a way for the machine to recognize the public key of a host from a previously established ssh connection. When you make connections to ssh servers, the fingerprints of every server you connect to is in your known_hosts file in ~/.ssh. The key in the known_hosts file didn't match the key that "Github" gave me. Github had [recently changed their ssh RSA host key](https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/) for security reasons, so maybe this was an opportunity of attack.

> One popular Tailscale feature is using another machine in your tailnet as an [exit node](https://tailscale.com/kb/1103/exit-nodes/)—an easy way to forward all internet traffic through another device. Luckily I had a Linux box at home on my tailnet that was already configured to be an exit node. I tried accessing Github using ssh with the Linux box, and it connected with no problem. This means what caused this ssh issue is related to not Github or how I configured my ssh keys, but the wifi network I was using.

Reviewer: ```suggestion
I needed to figure out whether the issue was with Github's key or with my connection. To do so, I turned to a popular Tailscale feature: [exit nodes](https://tailscale.com/kb/1103/exit-nodes/), which allow you to forward all your internet traffic through another machine in your tailnet. Luckily I had a Linux box at home on my tailnet that was already configured to be an exit node. I tried accessing Github using ssh with the Linux box, and it connected with no problem. This means that what caused this ssh issue is related not to Github or how I configured my ssh keys, but to the wifi network I was using.
```

Sorry I know we'd gone back and forth on this paragraph but seeing it here gave me a thought on how to make it clearer

Most VPNs let you route traffic through another computer, but not always one that you control (and that has your known_hosts file saved). What makes using Tailscale even more compelling in this scenario is, to improve bandwidth, I can configure my exit node to only advertise routes to github.com–what we call “subnet routers”.

For example, here's how you can advertise the routes for GitHub's git ingress servers:

```

> $ tailscale set –-advertise-routes=”$(curl -fsSL https://api.github.com/meta | jq -r ‘.git[]’ | tr ‘\n’ ‘,’)

Reviewer: ```suggestion
$ tailscale set –-advertise-routes="$(curl -fsSL https://api.github.com/meta | jq -r '.git[]' | tr '\n' ',')"
```

I think Google docs "smartened" the quotes here, right?

```

<Figure
  src="/images/on-the-node-while-on-the-road/exit-node-diagram.png"
  width={880}
  height={375}
  alt={`Blackpink in your area.`}
/>

> Here’s a diagram showing how I used my exit node. Blackpink is the linux box I used as an exit node, and keep in mind when using exit nodes if latency matters in your use case, you might want to use a machine that’s closer to you.

Reviewer: ```suggestion
Here’s a diagram showing how I used my exit node. Blackpink is the linux box I used as an exit node. Keep in mind when using exit nodes if latency matters in your use case, you might want to use a machine that’s closer to you.
```
